code/lib/cli/src/automigrate/fixes/new-frameworks/utils.ts

```diff
@@ -0,0 +1,88 @@
+import type { Preset, StorybookConfig } from '@storybook/types';
+
+export const packagesMap: Record<string, { webpack5?: string; vite?: string }> = {
+  '@storybook/react': {
+    webpack5: '@storybook/react-webpack5',
+    vite: '@storybook/react-vite',
+  },
+  '@storybook/preact': {
+    webpack5: '@storybook/preact-webpack5',
+    vite: '@storybook/preact-vite',
+  },
+  '@storybook/server': {
+    webpack5: '@storybook/server-webpack5',
+  },
+  '@storybook/ember': {
+    webpack5: '@storybook/ember',
+  },
+  '@storybook/angular': {
+    webpack5: '@storybook/angular',
+  },
+  '@storybook/vue': {
+    webpack5: '@storybook/vue-webpack5',
+    vite: '@storybook/vue-vite',
+  },
+  '@storybook/vue3': {
+    webpack5: '@storybook/vue3-webpack5',
+    vite: '@storybook/vue3-vite',
+  },
+  '@storybook/svelte': {
+    webpack5: '@storybook/svelte-webpack5',
+    vite: '@storybook/svelte-vite',
+  },
+  '@storybook/web-components': {
+    webpack5: '@storybook/web-components-webpack5',
+    vite: '@storybook/web-components-vite',
+  },
+  '@storybook/html': {
+    webpack5: '@storybook/html-webpack5',
+    vite: '@storybook/html-vite',
+  },
+};
+
+export const getBuilderInfo = (
+  mainConfig: StorybookConfig & { builder?: string | Preset }
+): { name: 'vite' | 'webpack5'; options: any } => {
+  let builderOptions = {};
+  let builderName = 'webpack5';
+
+  const { core = {}, framework } = mainConfig;
+  const { builder } = core;
+
+  if (builder) {
+    if (typeof builder === 'string') {
+      builderName = builder.includes('vite') ? 'vite' : 'webpack5';
```

Comment:
I'm not too fond of this inline condition for extensibility reasons. What if we introduce a new builder? This code will very likely break.

Instead let's do it like this:

```tsx
type Builders = 'vite' | 'webpack5';

export const getBuilderInfo = (...): { name: Builders; options: any } => {

  let builderName: Builders | null = null;
  
  if (builder) {
    if (typeof builder === 'string') {
      if(builder.includes('vite')) {
        builderName = 'vite'
      }
    ...
  if(builderName === null) {
    throw Error("Builder couldn't be extracted")
  }
}